### PR TITLE
Implement router

### PR DESCRIPTION
**Pull request by xepozz**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -2,18 +2,18 @@
...
-[![Total Downloads](https://poser.pugx.org/yiisoft/_____/downloads.png)](https://packagist.org/packages/yiisoft/_____)
-[![Build status](https://github.com/yiisoft/_____/workflows/build/badge.svg)](https://github.com/yiisoft/_____/actions?query=workflow%3Abuild)
-[![Code Coverage](https://codecov.io/gh/yiisoft/_____/branch/master/graph/badge.svg)](https://codecov.io/gh/yiisoft/_____)
-[![Mutation testing badge](https://img.shields.io/endpoint?style=flat&url=https%3A%2F%2Fbadge-api.stryker-mutator.io%2Fgithub.com%2Fyiisoft%2F_____%2Fmaster)](https://dashboard.stryker-mutator.io/reports/github.com/yiisoft/_____/master)
-[![static analysis](https://github.com/yiisoft/_____/workflows/static%20analysis/badge.svg)](https://github.com/yiisoft/_____/actions?query=workflow%3A%22static+analysis%22)
-[![type-coverage](https://shepherd.dev/github/yiisoft/_____/coverage.svg)](https://shepherd.dev/github/yiisoft/_____)
-[![psalm-level](https://shepherd.dev/github/yiisoft/_____/level.svg)](https://shepherd.dev/github/yiisoft/_____)
+[![Latest Stable Version](https://poser.pugx.org/yiisoft/File Router/v/stable.png)](https://packagist.org/packages/yiisoft/File Router)
```
Comment (samdark) [inline README.md]: Doesn't look correct.

File: src/FileRouter.php [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+        );
+
+        if (class_exists($className)) {
+            return $className;
+        }
+
+        // alternative version finding namespace by file
+
+
+        //$controllerDirectory = $this->aliases->get('src') . DIRECTORY_SEPARATOR . $this->baseControllerDirectory;
```
Comment (samdark) [inline src/FileRouter.php]: Remove?

File: src/FileRouter.php [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    }
+
+    private function parseAction(ServerRequestInterface $request): ?string
+    {
+        $action = match ($request->getMethod()) {
+            'HEAD', 'GET' => 'index',
+            'POST' => 'create',
+            'PUT' => 'update',
+            'DELETE' => 'delete',
+            default => throw new Exception('Not implemented.'),
```
Comment (samdark) [inline src/FileRouter.php]:
```suggestion
            default => throw new \Exception('Not implemented.'),
```

File: src/FileRouter.php [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+    {
+        $path = $request->getUri()->getPath();
+        if ($path === '/') {
+            $controllerName = 'Index';
+            $directoryPath = '';
+        } else {
+            $controllerName = preg_replace_callback(
+                '#(/.)#',
+                fn(array $matches) => strtoupper($matches[1]),
+                str_replace('/', DIRECTORY_SEPARATOR, $path)
```
Comment (Tigrov) [inline src/FileRouter.php]:
```suggestion
                $path
```

File: src/FileRouter.php (hunk 2 above)
Comment (xepozz) [reply src/FileRouter.php]: https://github.com/yiisoft/file-router/issues/2

File: README.md [hunk 5]
```diff
@@ -26,7 +26,7 @@ The package ...
 The package could be installed with composer:
 
 ```shell
-composer require yiisoft/_____
+composer require yiisoft/File Router
```
Comment (samdark) [inline README.md]:
```suggestion
composer require yiisoft/file-router
```

**Review by samdark — APPROVED** [review]

**Pull request merged**